### DISCUSSION TITLE
paddleocr  Out of memory error

### DISCUSSION THREAD

**Discussion by kennyLSN** [PaddlePaddle/PaddleOCR#12956] [Q&A]
#### 问题描述 / Problem Description
paddleocr 做长图的ocr识别，det_limit_side_len=20000，10G 显存，Out of memory error on 

#### 运行环境 / Runtime Environment
- OS:
- Paddle: paddlepaddle-gpu==2.6.0
- PaddleOCR:

NVIDIA-SMI 470.57.02    Driver Version: 470.57.02    CUDA Version: 11.8     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA A10          On   | 00000000:00:09.0 Off |                    0 |
|  0%   52C    P0    64W / 150W |  10262MiB / 10332MiB |      1%      Default |
|                               |                      |                  N/A |

#### 完整报错 / Complete Error Message

7   phi::DnnWorkspaceHandle::RunFunc(std::function<void (void*)> const&, unsigned long)
8   phi::DnnWorkspaceHandle::ReallocWorkspace(unsigned long)
9   paddle::memory::allocation::Allocator::Allocate(unsigned long)
10  paddle::memory::allocation::StatAllocator::AllocateImpl(unsigned long)
11  paddle::memory::allocation::Allocator::Allocate(unsigned long)
12  paddle::memory::allocation::Allocator::Allocate(unsigned long)
13  paddle::memory::allocation::Allocator::Allocate(unsigned long)
14  paddle::memory::allocation::Allocator::Allocate(unsigned long)
15  paddle::memory::allocation::CUDAAllocator::AllocateImpl(unsigned long)
16  std::string phi::enforce::GetCompleteTraceBackString<std::string >(std::string&&, char const*, int)
17  phi::enforce::GetCurrentTraceBackString[abi:cxx11](bool)

----------------------
Error Message Summary:
----------------------
ResourceExhaustedError: 

Out of memory error on GPU 0. Cannot allocate 128.000000MB memory on GPU 0, 10.021484GB memory has been allocated and available memory is only 70.000000MB.

Please check whether there is any other process using GPU 0.
1. If yes, please stop them, or start PaddlePaddle on another GPU.
2. If no, please decrease the batch size of your model. 
 (at /paddle/paddle/fluid/memory/allocation/cuda_allocator.cc:86)

**Comment by UserWangZz** [comment]
det_limit_side_len=20000设置的太大了，显存不够

**Comment by GreatV** [comment]
长图可以试试main分支下的slice操作

https://github.com/PaddlePaddle/PaddleOCR/blob/6954da712e1cb7dec913672a9a7ffee78ea2293b/doc/doc_ch/quickstart.md?plain=1#L256-L294

**Comment by kennyLSN** [comment]
> 长图可以试试main分支下的slice操作
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/6954da712e1cb7dec913672a9a7ffee78ea2293b/doc/doc_ch/quickstart.md?plain=1#L256-L294



> 长图可以试试main分支下的slice操作
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/6954da712e1cb7dec913672a9a7ffee78ea2293b/doc/doc_ch/quickstart.md?plain=1#L256-L294

请问这是啥问题。。
/paddleocr.py", line 728, in ocr
    dt_boxes, rec_res, _ = self.__call__(img, cls, slice)
TypeError: __call__() takes from 2 to 3 positional arguments but 4 were given

**Comment by GreatV** [comment]
我又试了一下没啥问题呀

```shell
pip install git+https://github.com/PaddlePaddle/PaddleOCR.git
```

```Python
from paddleocr import PaddleOCR 
from PIL import Image, ImageDraw, ImageFont 

# 初始化OCR引擎 
ocr = PaddleOCR(use_angle_cls=True, lang="en") 

img_path = "./very_large_image.jpg" 
slice = {'horizontal_stride': 300, 'vertical_stride': 500, 'merge_x_thres': 50, 'merge_y_thres': 35} 
results = ocr.ocr(img_path, cls=True, slice=slice) 

# 加载图像 
image = Image.open(img_path).convert("RGB") 
draw = ImageDraw.Draw(image) 
font = ImageFont.truetype("../PaddleOCR/doc/fonts/simfang.ttf", size=20)  # 根据需要调整大小 

# 处理并绘制结果 
for res in results: 
    for line in res: 
        box = [tuple(point) for point in line[0]] 
        # 找出边界框 
        box = [(min(point[0] for point in box), min(point[1] for point in box)), 
            (max(point[0] for point in box), max(point[1] for point in box))] 
        txt = line[1][0] 
        draw.rectangle(box, outline="red", width=2)  # 绘制矩形 
        draw.text((box[0][0], box[0][1] - 25), txt, fill="blue", font=font)  # 在矩形上方绘制文本 

# 保存结果 
image.save("result.jpg") 
```
![result](https://github.com/PaddlePaddle/PaddleOCR/assets/17264618/8bb011a0-10b1-40d0-a7a7-b7ea019f5198)
